outlines/models/base.py

```diff
@@ -51,6 +51,41 @@ class Model(ABC):
 
     type_adapter: ModelTypeAdapter
 
+    def __init__(self, tensor_library_name=None):
+        """Initialize the model.
+
+        Parameters
+        ----------
+        tensor_library_name: Optional name of the tensor library to use.
+            If not provided, the default for the model will be used.
+
+        """
+        self._tensor_library_name = tensor_library_name
+
+    @property
+    def tensor_library_name(self):
+        """Return the name of the tensor library used by the model.
+
+        If a custom tensor library name was provided during initialization,
+        that value will be returned. Otherwise, the model's default tensor
+        library name will be used.
+        """
+        if self._tensor_library_name is not None:
+            return self._tensor_library_name
+
+        return self._default_tensor_library_name
+
+    @property
+    def _default_tensor_library_name(self):
```

Comment:
What is the idea behind having methods instead of attributes?

Reply:
I thought it's a clearer way of documenting in the base class what's expected of child classes (and implementation is more straightforward there). Otherwise we would need to add something like that in the child classes:
```python
  def __init__(self, tensor_library_name):
      self._tensor_library_name = tensor_library_name or "torch"
```
But I thought it may be harder to understand that there's this attribute for local models and that the associated classes need to add that line

Reply:
Can't you just specify `_tensor_library_name: str` under the class name?

Reply:
I thought mypy would not like it for some reason, my bad.